app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,64 @@
+class SessionsController < ApplicationController
+  skip_before_action :redirect_guest_users, except: :welcome
+  before_action :redirect_current_user, only: [:new, :login, :reset_password_form, :reset_password_form]
+  before_action :get_user_by_email, only: [:forgot_password, :login]
+  before_action :get_user_by_id, only: [:reset_password_form, :reset_password]
+  before_action :check_user_validity, only: :login
+
+  def new
+    @user = User.new
+  end
+
+  def login
+    remember_me?
+    redirect_unverified_user
+    session[:api_token] = @user.api_token
+    redirect_to welcome_path
+  end
+
+  def redirect_unverified_user
+    redirect_to login_url, notice: t('.unverified') unless @user.verified
+  end
+
+  #for logout
+  def destroy
+    logout
+  end
+
+  def forgot_password
+    if @user.present?
+      @user.set_forgot_password_token
+    else
+      redirect_to login_path, notice: t('.user_not_present')
+    end
+  end
+
+  def reset_password_form
+    render 'invalid_url' unless user.present? && @user.forgot_password_token == params[:token]
+  end
+
+  def reset_password
+    if password_matched?
+      @user.update_password(params[:password])
+      @user.set_api_token
```

Comment:
@user is not saved in `set_api_token`. Check this.

Reply:
Since every time a password is changed, we need to reset the api token to logout user from all the devices. Call `set_api_token` method from update password.